src/gc.h

```diff
@@ -147,33 +143,90 @@ typedef struct {
     uint8_t *ages;
 } jl_gc_pagemeta_t;
 
-typedef struct {
-    char data[GC_PAGE_SZ];
-} jl_gc_page_t
-#if !defined(_COMPILER_MICROSOFT_) && !(defined(_COMPILER_MINGW_) && defined(_COMPILER_CLANG_))
-__attribute__((aligned(GC_PAGE_SZ)))
+// Page layout:
+//  Newpage freelist: sizeof(void*)
+//  Padding: GC_PAGE_OFFSET - sizeof(void*)
+//  Blocks: osize * n
+//    Tag: sizeof(jl_taggedvalue_t)
+//    Data: <= osize - sizeof(jl_taggedvalue_t)
+
+// Memory map:
+//  The complete address space is divided up into a multi-level page table.
+//  The three levels have similar but slightly different structures:
+//    - pagetable0_t: the bottom/leaf level (covers the contiguous addresses)
+//    - pagetable1_t: the middle level
+//    - pagetable2_t: the top/leaf level (covers the entire virtual address space)
+//  Corresponding to these simlar structures is a large amount of repetitive
```

Comment:
Those are debugging functions not actually part of he gc. They are usually commented out, so I wouldn't worry about them.